### PR TITLE
Core/removing default comm in sparse matrix

### PR DESCRIPTION
**Pull request by RiccardoRossi**
**Description**
title tells it all.

Please mark the PR with appropriate tags: 
- mpi

**Changelog**
now the communicator needs to be passed explicitly to the distributed_sparse_graph

### CODE REVIEW

File: kratos/containers/distributed_csr_matrix.h [hunk 1]
```diff
@@ -963,7 +963,9 @@ class DistributedCsrMatrix
 
     void load(Serializer& rSerializer)
     {
-        rSerializer.load("CSRcommunicator",mrComm);
+        std::string comm_name;
+        rSerializer.load("CommunicatorName",comm_name);
+        mrComm = ParallelEnvironment::GetDataCommunicator(comm_name);
```
Comment (philbucher) [inline kratos/containers/distributed_csr_matrix.h]: hm I am not sure if this will work
maybe you need to use a pointer after all

Comment (RiccardoRossi) [reply kratos/containers/distributed_csr_matrix.h]: yes, done.
 strange thing is that it did compile locally (i don't understand how though...)

**Review by philbucher — DISMISSED** [review]
:+1:

**Review by philbucher — DISMISSED** [review]

File: kratos/containers/distributed_csr_matrix.h [hunk 2]
```diff
@@ -70,20 +70,25 @@ class DistributedCsrMatrix
     ///@}
     ///@name Life Cycle
     ///@{
-    DistributedCsrMatrix(const DataCommunicator& rComm=ParallelEnvironment::GetDefaultDataCommunicator()) //default constructor. To be used for low level operations
-        : mrComm(rComm)
+
+    //default constructor. To be used for low level operations
+    DistributedCsrMatrix() 
+    {}
```
Comment (philbucher) [inline kratos/containers/distributed_csr_matrix.h:76]: if this is only used for the serialization then I would move it to `private`

Comment (RiccardoRossi) [reply kratos/containers/distributed_csr_matrix.h:76]: no, this can also be used in very low level routines, like when converting to amgcl

**Comment by RiccardoRossi** [comment]
@philbucher could u reapprove?

**Review by philbucher — APPROVED** [review]

**Pull request merged**